### PR TITLE
(SIMP-4443) Document Difference between simp and true in

### PR DESCRIPTION
**Pull request by jeannegreulich**
simp_options::pki
- updated documentation
- consolidate FakeCA documentation in one place.
- remove comment about auto for gencerts because it doesn't do anything.

SIMP-4443 #close
SIMP-4474 #close
SIMP-4475 #close

### CODE REVIEW

File: docs/user_guide/Certificates/Certificates.inc [hunk 1]
```diff
@@ -1,10 +1,10 @@
 Apply Certificates
 ==================
 
-All clients in a SIMP system must have :term:`Public Key Infrastructure` (PKI)
-keypairs generated for the server.  These keys reside in the
-``/var/simp/environments/simp/site_files/pki_files/files/keydist`` directory on the
-SIMP server and are served to the clients over the puppet protocol.
+All clients in a SIMP system should have :term:`Public Key Infrastructure` (PKI)
```
Comment (lnemsick-simp) [inline docs/user_guide/Certificates/Certificates.inc]: why was this changed from 'should' to 'must'?

File: docs/user_guide/Certificates/Certificates.inc [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
+keypairs generated for the server.  These are the refered to as the infrastructure
```
Comment (lnemsick-simp) [inline docs/user_guide/Certificates/Certificates.inc]: 'refered' -> 'referred'

File: docs/user_guide/Certificates/Certificates.inc [hunk 3]
```diff
@@ -13,6 +13,25 @@ SIMP server and are served to the clients over the puppet protocol.
 
    See :ref:`Certificate Management` for more information.
 
-This section provides guidance on installing official certificates or, as an
-interim measure, generating certificates from the Fake (self-signing)
-Certificate Authority provided by SIMP.
+SIMP uses the ``pupmod-simp-pki`` module to help distribute keypairs.
```
Comment (lnemsick-simp) [inline docs/user_guide/Certificates/Certificates.inc]: 'distribute keypairs' -> 'distribute infrastructure keypairs'

File: docs/user_guide/Certificates/Certificates.inc [hunk 4]
```diff
@@ -13,6 +13,25 @@ SIMP server and are served to the clients over the puppet protocol.
...
+The global variable, ``simp_options::pki`` determines what parts of the module are
+included.  It can be overridden at several levels if different hosts or
```
Comment (lnemsick-simp) [inline docs/user_guide/Certificates/Certificates.inc]: 'can be overridden' -> 'can be overridden in hieradata'

File: docs/user_guide/Certificates/Certificates.inc [hunk 5]
```diff
@@ -13,6 +13,25 @@ SIMP server and are served to the clients over the puppet protocol.
...
+applications need to hande certificates differently.
+
+``simp_options::pki`` can have one of three settings:
+
```
Comment (lnemsick-simp) [inline docs/user_guide/Certificates/Certificates.inc]: Numbered items aren't formatted quite right.  Need to indent continuation lines.

File: docs/user_guide/Certificates/Certificates.inc [hunk 6]
```diff
@@ -13,6 +13,25 @@ SIMP server and are served to the clients over the puppet protocol.
...
+1) ``simp`` -  Keypairs are distributed from a central location on the puppetserver to
+the ``/etc/pki/simp/x509`` directory on the client. Any applications using them will
+then make a copy in ``/etc/pki/simp_apps/<app name>/x509`` with the correct permissions
+for an application to use.
+
+2) ``true`` -  Applicatons on the clients will copy the keypairs from a local directory
```
Comment (lnemsick-simp) [inline docs/user_guide/Certificates/Certificates.inc]: 'Applicatons' -> 'SIMP-managed applications'

File: docs/user_guide/Certificates/Certificates.inc [hunk 7]
```diff
@@ -13,6 +13,25 @@ SIMP server and are served to the clients over the puppet protocol.
...
+on the client to ``/etc/pki/simp_apps/<app name>/x509``. The default local directory to
+copy from is ``/etc/pki/simp/x509`` but this can be overidden by setting the ``simp_options::pki::source``
+variable.
+
+3) ``false`` -  The user will have to manage keypairs themselves. You will need to look at
+each module that uses pki on a client to determine what variables need to be set.
+NOTE: This does not disable the use of pki in a module.
```
Comment (lnemsick-simp) [inline docs/user_guide/Certificates/Certificates.inc]: * This should be formatted as a NOTE within item 3.  That way it will stand out and still be associated with the false description.
* Should it be WARNING, instead?  Seems to have confused users...

File: docs/user_guide/Certificates/Official_Certificates.inc [hunk 8]
```diff
@@ -1,38 +1,54 @@
...
 --------------------------------
 
-Below are the steps to install official certificates for a SIMP client on
-the SIMP server:
+This section describes how to install infrastructure certificates from an
+official certificate authority on the puppetserver for distribution to client
+servers.  You need to have simp_options::pki set to ``simp`` on the client for
+this to work.
+
+The key distribution directory on the puppetserver is located under the site_files directory.
```
Comment (lnemsick-simp) [inline docs/user_guide/Certificates/Official_Certificates.inc]: site_files instances need to be double back quoted just like other references to directories.

File: docs/user_guide/Certificates/Official_Certificates.inc [hunk 9]
```diff
@@ -1,38 +1,54 @@
...
+The site_files directory is an alternate module path set in each environment's environment.conf file.
+By default the site_files directory is ``/var/simp/environments/simp/site_files/``
+and the key distribution directory is ``pki_files/files/keydist``.  The system expects each client
```
Comment (lnemsick-simp) [inline docs/user_guide/Certificates/Official_Certificates.inc]: * 'directory' -> 'sub-directory'
* The second simp in the path, environments/<simp>/site_files, is the environment.  Should we have a description of this nuance here?

File: docs/user_guide/Certificates/Official_Certificates.inc [hunk 10]
```diff
@@ -1,38 +1,54 @@
...
+to have a directory that is its FQDN and the certs should be named with the FQDN.
```
Comment (lnemsick-simp) [inline docs/user_guide/Certificates/Official_Certificates.inc]: https://github.com/simp/pupmod-simp-pki/pull/45 allows users to override this.  Should this description reflect this new option, or would that be too much detail?

File: docs/user_guide/Certificates/Official_Certificates.inc [hunk 11]
```diff
@@ -1,38 +1,54 @@
...
+
+To install official certificates on the puppetserver do the following:
 
 1. Copy the certificates received from a proper :term:`CA` to the SIMP server.
-2. Add the keys for the node to ``/var/simp/environments/simp/site_files/pki_files/files/keydist``.
+2. Add the certificates for the node to key disttibution directory in the site_files.
```
Comment (lnemsick-simp) [inline docs/user_guide/Certificates/Official_Certificates.inc]: * 'disttibution' -> 'distribution'
* I preferred the fully-qualified path, as it would be easy to cut and paste.  Was it removed to allow for a different environment?

File: docs/user_guide/Certificates/Official_Certificates.inc [hunk 12]
```diff
@@ -1,38 +1,54 @@
...
+
+   a) Make the directory  under the key distribution directory for the clients certificates using the clients FQDN.
+   b) Copy the official certificate to that directory using the FQDN and the correct extension.
```
Comment (lnemsick-simp) [inline docs/user_guide/Certificates/Official_Certificates.inc]: List the correct extension?

File: docs/user_guide/Certificates/Official_Certificates.inc [hunk 13]
```diff
@@ -1,38 +1,54 @@
...
+   c) Make sure the permissions are correct.
```
Comment (lnemsick-simp) [inline docs/user_guide/Certificates/Official_Certificates.inc]: Indicate what should the permissions be?

File: docs/user_guide/Certificates/Official_Certificates.inc [hunk 14]
```diff
@@ -1,38 +1,54 @@
...
-  a) Type ``mkdir -p /var/simp/environments/simp/site_files/pki_files/files/keydist/***<Client System FQDN>***``
-  b) Type
+   For example to install certificates for a system named mycomputer.my.domain:
 
-     .. code-block:: bash
+   .. code-block:: shell
 
-       mv ***<Certificate Directory>***/***<FQDN>***.[pem|pub] \
-       /var/simp/environments/simp/site_files/pki_files/files/keydist/***<FQDN>***
+     mkdir -p /var/simp/environments/simp/site_files/pki_files/files/keydist/mycomuter.my.domain
```
Comment (lnemsick-simp) [inline docs/user_guide/Certificates/Official_Certificates.inc]: * my_environment=simp ?
* '/var/simp/environments/simp/site_files' -> '/var/simp/environments/${my_environment}/site_files'
* 'mycomuter' -> 'mycomputer' in a couple of places

File: docs/user_guide/Certificates/Official_Certificates.inc [hunk 15]
```diff
@@ -1,38 +1,54 @@
...
+     mv /dir/where/the/certs/were/myprivatecert.pem /var/simp/environments/simp/site_files/pki_files/files/keydist/mycomupter.my.domain.pem
+     mv /dir/where/the/certs/were/mypubliccert.pub /var/simp/environments/simp/site_files/pki_files/files/keydist/mycomupter.my.domain.pub
+     chown -R root.puppet /var/simp/environments/simp/site_files/pki_files/files/keydist
+     chmod -R u=rwX,g=rX,o-rwx /var/simp/environments/simp/site_files/pki_files/files/keydist
```
Comment (lnemsick-simp) [inline docs/user_guide/Certificates/Official_Certificates.inc]: Do we need to set the selinux context?

File: docs/user_guide/Certificates/Official_Certificates.inc [hunk 16]
```diff
@@ -1,38 +1,54 @@
...
-  c) Type ``cd cacerts``
-  d) Type ``for file in *.pem; do ln -s $file `openssl x509 -in $file -hash -noout`.0; done``
+   .. code-block:: shell
+
+     cd /var/simp/environments/simp/site_files/pki_files/files/keydist
+     mkdir cacerts
+     cd cacerts
+     for file in *.pem; do ln -s $file `openssl x509 -in $file -hash -noout`.0; done
 
 Generating Certificates from the Fake CA
```
Comment (lnemsick-simp) [inline docs/user_guide/Certificates/Official_Certificates.inc]: Should this be 'Generating Infrastructure Certificates fro the Fake CA'?

File: docs/user_guide/Certificates/Official_Certificates.inc [hunk 17]
```diff
@@ -1,38 +1,54 @@
...
 ----------------------------------------
 
-If server certificates have not or could not be obtained at the time of client
-installation, SIMP provides a way to create them for the system, so that it
-will work until proper certificates are provided.
+The Fake (self signing) Certificate Authority (Fake CA) is provided by SIMP
+as a way to obtain server certificates if officail certificates could not be
+obtained at the time of client installation or in testing environments.
```
Comment (lnemsick-simp) [inline docs/user_guide/Certificates/Official_Certificates.inc]: 'or in testing environments' -> 'or the servers are operating in a test environment'

File: docs/user_guide/SIMP_Administration/General_Administration/Certificate_Management.inc [hunk 18]
```diff
@@ -4,71 +4,51 @@ Certificate Management
...
+Infrastructure Certificates
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Infrastructure certificates, commonly refered to as server certificates,
+are the standard :term:`PKI` certificates assigned by an official :term:`CA`.
+These are used by all other applications that require certifcates.
+
+SIMP offers capabilities to help manage these certificates including a FakeCA
+utility that will provide self-signed certificates for use  on testing systems
+or until official certifcates can be obtained.
```
Comment (lnemsick-simp) [inline docs/user_guide/SIMP_Administration/General_Administration/Certificate_Management.inc]: 'certifcates' -> 'certificates'

File: docs/user_guide/Certificates/Certificates.inc [hunk 19]
```diff
@@ -13,6 +13,25 @@ SIMP server and are served to the clients over the puppet protocol.
...
-Certificate Authority provided by SIMP.
+SIMP uses the ``pupmod-simp-pki`` module to help distribute keypairs.
+The global variable, ``simp_options::pki`` determines what parts of the module are
+included.  It can be overridden at several levels if different hosts or
+applications need to hande certificates differently.
+
+``simp_options::pki`` can have one of three settings:
+
+1) ``simp`` -  Keypairs are distributed from a central location on the puppetserver to
+the ``/etc/pki/simp/x509`` directory on the client. Any applications using them will
```
Comment (lnemsick-simp) [inline docs/user_guide/Certificates/Certificates.inc]: 'Any applications' -> 'Any SIMP-managed applications'

File: docs/user_guide/SIMP_Administration/General_Administration/Certificate_Management.inc [hunk 20]
```diff
@@ -4,71 +4,51 @@ Certificate Management
...
-The following is an example of how to change the expiration time from one year
-(the default) to five years for any newly created certificate.
-
-.. code-block:: bash
-
-   for file in $(grep -rl 365 /var/simp/environments/simp/FakeCA/)
-   do
-      sed -i 's/365/1825/' $file
-   done
+and provides links to further information.
```
Comment (lnemsick-simp) [inline docs/user_guide/SIMP_Administration/General_Administration/Certificate_Management.inc]: Seems like this introductory section should also be in the User Guide Apply Certificates.  Logically, the user won't get to this page until after they went through the client install.  So, they won't get this helpful explanation until after they went through the details of setting up infrastructure certs (which in the middle of it references Puppet certs).  Very confusing IMHO.

**Comment by trevor-vaughan** [comment]
@jeannegreulich Looks like we need to update some links as well https://travis-ci.org/simp/simp-doc/jobs/348336238

File: docs/user_guide/Certificates/Official_Certificates.inc (hunk 12 above)
Comment (judyj) [reply docs/user_guide/Certificates/Official_Certificates.inc]: client's FQDN (apostrophe)

File: docs/user_guide/SIMP_Administration/General_Administration/Certificate_Management.inc (hunk 20 above)
Comment (jeannegreulich) [reply docs/user_guide/SIMP_Administration/General_Administration/Certificate_Management.inc]: Entered ticket SIMP-4159 with points on reconfiguring the certificate section among the data/

**Comment by lnemsick-simp** [comment]
Working on a few fixes that were missed in the update.

**Comment by jeannegreulich** [comment]
Liz's fixes look fine to me.
